src/Notification/NotificationHandler.php

```diff
@@ -86,30 +94,61 @@ public function receiveNotification(ReceiveNotification $wrapper): ReceiveNotifi
-                    $event = new StatusChangeOngoingEvent($product, $product->getAcceptedDeadline());
-                }
-                if ($product->getStatus() === self::PRODUCT_STATUS_REQUESTED) {
-                    $event = new StatusChangeRequestedEvent($product);
+                switch ($product->getStatus()) {
+                    case self::PRODUCT_STATUS_CHANGE_REQUESTED:
+                        $event = new Product\StatusChangeRequestedEvent($product);
+                        break;
+                    case self::PRODUCT_STATUS_CHANGE_ACCEPTED:
+                        $event = new Product\StatusChangeAcceptedEvent($product);
+                        break;
+                    case self::PRODUCT_STATUS_CHANGE_ONGOING:
+                        $event = new Product\StatusChangeOngoingEvent($product, $product->getAcceptedDeadline());
+                        break;
+                    case self::PRODUCT_STATUS_CHANGE_READY_TO_BE_SENT:
+                        $event = new Product\StatusChangeReadyToBeSentEvent($product);
+                        break;
+                    case self::PRODUCT_STATUS_CHANGE_SENT:
+                        $event = new Product\StatusChangeSentEvent($product);
+                        break;
+                    case self::PRODUCT_STATUS_CHANGE_CANCELLED:
+                        $event = new Product\StatusChangeCancelledEvent($product);
+                        break;
+                    case self::PRODUCT_STATUS_CHANGE_CLOSED:
+                        $event = new Product\StatusChangeClosedEvent($product);
+                        break;
+                    case self::PRODUCT_STATUS_CHANGE_SUSPENDED:
+                        $event = new Product\StatusChangeSuspendedEvent($product);
+                        break;
                 }
                 break;
             case self::NOTIFICATION_PRODUCT_DELIVERY:
-                $event = new DeliveryEvent($notification->getProduct());
+                $event = new Product\DeliveryEvent($notification->getProduct());
                 break;
             case self::NOTIFICATION_REQUEST_STATUS_CHANGE:
                 $request = $notification->getLinguisticRequest();
-                if ($request->getStatus() === self::REQUEST_STATUS_ACCEPTED) {
-                    $event = new ChangeAcceptedEvent($request, $notification->getPlanningAgent(), $notification->getPlanningSector());
-                }
-                if ($request->getStatus() === self::REQUEST_STATUS_REJECTED) {
-                    $event = new ChangeRejectedEvent($request, $notification->getPlanningAgent(), $notification->getPlanningSector(), $notification->getMessage());
+                switch ($request->getStatus()) {
+                    case self::REQUEST_STATUS_CHANGE_ACCEPTED:
+                        $event = new Request\StatusChangeAcceptedEvent($request, $notification->getPlanningAgent(), $notification->getPlanningSector(), $notification->getMessage());
```

Comment:
The request status change accepted *requires* a message? The example doc doesn't have a message. The rejected one has a message.